### DISCUSSION TITLE
Is it possible to have readable friendly css variable names?

### DISCUSSION THREAD

**Discussion by ShaneYu** [facebook/stylex#667] [Q&A]
I love the use of defineVars and create to build my styles, it's excellent, however the resulting CSS output has randomly generated CSS variables names that mean nothing to anyone looking at them; whilst this is okay 99% of the time, I wonder is it possible to have those variables names something easier to read?

For example rather than `--x3s7y4r: #991b1b;` I'd like to see `--alert-danger-bg: #991b1b;`.

This would be useful for when our UX designer wants to tweak these variables in the browser on the fly and see the impact of those changes.

**Comment by ShaneYu** [comment]
> I love the use of defineVars and create to build my styles, it's excellent, however the resulting CSS output has randomly generated CSS variables names that mean nothing to anyone looking at them; whilst this is okay 99% of the time, I wonder is it possible to have those variables names something easier to read?
> 
> 
> 
> For example rather than `--x3s7y4r: #991b1b;` I'd like to see `--alert-danger-bg: #991b1b;`.
> 
> 
> 
> This would be useful for when our UX designer wants to tweak these variables in the browser on the fly and see the impact of those changes.

Don't know how I missed it, another discussion exists for this already here: https://github.com/facebook/stylex/discussions/656

**Comment by nmn** [comment]
There is a new feature in StyleX where you can use `'--name'` as keys for your variables in StyleX and StyleX will keep the literals as-in instead of generating a hashed name for it.
